packages/element/src/lib/element.ts

```diff
@@ -1,55 +1,60 @@
 import { metadataStore } from './metadata.js';
 
-export function element<Target extends CustomElementConstructor>(
-  Base: Target,
-  ctx: ClassDecoratorContext<Target>
-) {
-  const meta = metadataStore.read(ctx.metadata);
-
-  ctx.addInitializer(function (this: Target) {
-    if (meta.tagName) {
-      const val = meta.tagName(this);
-
-      if (!customElements.get(val)) {
-        customElements.define(val, this);
+export interface ElementOpts {
+  tagName?: string;
+}
+
+export function element(opts?: ElementOpts) {
+  return <Target extends CustomElementConstructor>(
+    Base: Target,
+    ctx: ClassDecoratorContext<Target>
+  ) => {
+    const meta = metadataStore.read(ctx.metadata);
+
+    ctx.addInitializer(function (this: Target) {
+      if (opts?.tagName) {
```

Comment:
Oh right it's optional to use tagName to autoregister in customElements, that kept me from trying this out so far...
Gotta give `@element()/@attr()` a try as soon as I get back to my browsergame :D 🚀